spinflip/spin_flip_tools.py

```diff
@@ -129,24 +171,29 @@ def __init__(self, t_sim, emu_file, merger_data_loc, location, p_abs, resonance_
         
         #Gradients
         if gradient_filename != None:
-            # flux gradient (spacetime up, spacetime (gradient) low, F, F)
+            #check location and timestep are match gradients file
             all_gradJ, x, y, z, it, limits = read_gradients(gradient_filename)
-            assert(t_sim == it)
-            assert(location[0] >= limits[0,0] and location[0] <= limits[0,1]) # make sure the location is within the limits
+            self.Gradients_class = Gradients(gradient_filename, merger_data_loc)
```

Comment:
The class is the abstract definition. What you have here is an instance of that class. I would call it self.gradients or self.Gradients_instance